src/Libraries/RevitNodes/Application/Document.cs

```diff
@@ -62,6 +66,171 @@ public static Document Current
+        {
+            List<Autodesk.Revit.DB.ElementId> materials = new FilteredElementCollector(this.InternalDocument).OfClass(typeof(Autodesk.Revit.DB.Material))
+                                                                                                           .ToElementIds()
+                                                                                                           .ToList();
+
+            List<Autodesk.Revit.DB.Element> elementTypes = new FilteredElementCollector(this.InternalDocument).WhereElementIsElementType()
+                                                                                                              .ToElements()
+                                                                                                              .ToList();
+            var nonPurgedElements = new List<Autodesk.Revit.DB.ElementId>();
+
+            for (int i = 0; i < elementTypes.Count; i++)
+            {
+                HostObjAttributes hostObj = elementTypes[i] as HostObjAttributes;
+                if (hostObj != null)
+                {
+                    var compundStructure = hostObj.GetCompoundStructure();
+                    if (compundStructure == null)
+                        continue;
+
+                    int layerCount = compundStructure.LayerCount;
+                    for (int j = 0; j < layerCount; j++)
+                    {
+                        ElementId elementId = compundStructure.GetMaterialId(j);
+                        if (materials.Contains(elementId))
+                        {
+                            nonPurgedElements.Add(elementId);
+                            materials.Remove(elementId);
+                        }
+                        
+                    }
+                    continue;
+                }
+
+                List<ElementId> elementMaterialIds = elementTypes[i].GetMaterialIds(false).ToList();
+                if (elementMaterialIds.Any())
+                {
+                    materials.RemoveAll(purgedId => elementMaterialIds.Exists(elemId => purgedId == elemId));
+                    nonPurgedElements.AddRange(elementMaterialIds.Where(elmeMatId => !nonPurgedElements.Any(nonPurgedElemeMatId => nonPurgedElemeMatId == elmeMatId)));
+                }
+
+                if (materials == null)
+                    break;
+            }
+            if (materials.Count > 0)
+                this.InternalDocument.Delete(materials);
+            List<ElementId> purgedMaterials = new List<ElementId>();
+            purgedMaterials.AddRange(PurgeMaterialAssets(nonPurgedElements));
+            purgedMaterials.AddRange(materials);
+
+            return purgedMaterials;
+        }
+
+        private List<Autodesk.Revit.DB.ElementId> PurgeMaterialAssets(List<Autodesk.Revit.DB.ElementId> elementIds)
+        {
+            List<ElementId> appearanceAssets = new FilteredElementCollector(this.InternalDocument).OfClass(typeof(AppearanceAssetElement))
```

Comment:
name the vars more suggestively
```suggestion
            List<ElementId> appearanceAssetIds = new FilteredElementCollector(this.InternalDocument).OfClass(typeof(AppearanceAssetElement))
```